### PR TITLE
Convert docker-fpm-frr to buster

### PR DESCRIPTION
**Pull request by joyas-joseph**
Signed-off-by: Joyas Joseph <joyas_joseph@dell.com>

**- Why I did it**
This is an enhancement.

**- How I did it**
Code change

**- How to verify it**
Build `target/docker-fpm-frr.gz`
Verify that `/etc/apt/sources.list` points to `buster` using ` docker exec bgp cat /etc/apt/sources.list`

BGP neighborship  is established.
```
root@sonic:~# show ip bgp summary 

IPv4 Unicast Summary:
BGP router identifier 10.1.0.1, local AS number 65100 vrf-id 0
BGP table version 1
RIB entries 1, using 184 bytes of memory
Peers 1, using 20 KiB of memory

Neighbor        V         AS MsgRcvd MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd
6.1.1.1         4        100      96      96        0    0    0 01:32:04            0

Total number of neighbors 1
root@sonic:~#  

```

**- Description for the changelog**
Convert docker-fpm-frr to buster

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by pavel-shirshov** [comment]
I believe we can avoid building our own swig and libyang for buster.

**Comment by joyas-joseph** [comment]
> https://github.com/FRRouting/frr/blob/7bc33bcc1996b554328792d80a8e5557ccaba6f5/doc/developer/building-libyang.rst

patched libyang is required for mgmt-framework.

**Comment by pavel-shirshov** [comment]
@joyas-joseph what about swig? Can we use swig from the repo, and don't build it by ourselfes?

**Comment by joyas-joseph** [comment]
> @joyas-joseph what about swig? Can we use swig from the repo, and don't build it by ourselves?

Let me take a look.

**Comment by joyas-joseph** [comment]
retest broadcom please

**Comment by joyas-joseph** [comment]
retest vs please

**Comment by joyas-joseph** [comment]
retest broadcom please

**Comment by joyas-joseph** [comment]
retest vs please

File: build_image.sh [hunk 1]
```diff
@@ -150,7 +150,15 @@ elif [ "$IMAGE_TYPE" = "aboot" ]; then
...
-        [ -f "$CA_CERT" ] && cp "$CA_CERT" "$TARGET_CA_CERT"
+
+        # If the ca certificate does not exist, the test certificate will be used to sign the image
+        if [ ! -f "$CA_CERT" ]; then
+            TEST_CERT_PATH=files/image_config/secureboot/test-certs
+            CA_CERT="${TEST_CERT_PATH}/ca.cert"
+            SIGNING_KEY="${TEST_CERT_PATH}/signing.key"
+            SIGNING_CERT="${TEST_CERT_PATH}/signing.cert"
+        fi
+        cp "$CA_CERT" "$TARGET_CA_CERT"
```
Comment (pavel-shirshov) [inline build_image.sh:161]: Should it be part of another PR?

Comment (pavel-shirshov) [reply build_image.sh:161]: I see. Thanks

**Comment by joyas-joseph** [comment]
retest broadcom please

File: build_image.sh (hunk 1 above)
Comment (xumia) [reply build_image.sh:161]: It should be caused by force push in master branch, the cert had been removed.
@qiluo-msft, the change should not be correct, right?
ActionItem: If your PR contains that commit, then your PR will be impacted. You will need to re-adapt to the revised the git history after the commit is removed.

Comment (qiluo-msft) [reply build_image.sh:161]: Please refer https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!topic/sonicproject/aV8L81MCnpU

> If your PR contains that commit, then your PR will be impacted. You will need to re-adapt to the revised the git history after the commit is removed.

**Comment by lguohan** [comment]
@joyas-joseph , can you rebase this pr to the latest master.

File: sonic-slave-buster/Dockerfile.j2 [hunk 2]
```diff
@@ -305,6 +305,8 @@ RUN apt-get update && apt-get install -y \
 # For DHCP Monitor tool
         libexplain-dev \
         libevent-dev \
+# For libyang
+	swig \
```
Comment (qiluo-msft) [inline sonic-slave-buster/Dockerfile.j2:309]: tab -> spaces

File: sonic-slave-stretch/Dockerfile.j2 [hunk 3]
```diff
@@ -301,6 +301,8 @@ RUN apt-get update && apt-get install -y \
 # For DHCP Monitor tool
         libexplain-dev \
         libevent-dev \
+# For libyang
+	swig \
```
Comment (qiluo-msft) [inline sonic-slave-stretch/Dockerfile.j2:305]: tab -> spaces

**Comment by lguohan** [comment]
retest vs please

**Comment by lguohan** [comment]
it looks like there is some genuine build issue for vs.

**Comment by lguohan** [comment]
```
09:09:10  	install -d debian/libyang-cpp-dev//usr/lib/x86_64-linux-gnu/pkgconfig
09:09:10  	cp --reflink=auto -a debian/tmp/usr/lib/x86_64-linux-gnu/pkgconfig/libyang-cpp.pc debian/libyang-cpp-dev//usr/lib/x86_64-linux-gnu/pkgconfig/
09:09:10  dh_install: Cannot find (any matches for) "usr/lib/python3/dist-packages/*" (tried in ., debian/tmp)
09:09:10  
09:09:10  dh_install: python3-yang missing files: usr/lib/python3/dist-packages/*
09:09:10  dh_install: Cannot find (any matches for) "usr/lib/python2.7/dist-packages/*" (tried in ., debian/tmp)
09:09:10  
09:09:10  dh_install: python2-yang missing files: usr/lib/python2.7/dist-packages/*
09:09:10  dh_install: missing files, aborting
```

**Comment by joyas-joseph** [comment]
> it looks like there is some genuine build issue for vs.

Saw this in the logs.
```
-- Could NOT find SWIG: Found unsuitable version "3.0.10", but required is at least "3.0.12" (found /usr/bin/swig3.0)
```
Will revert the swig related change and come back to it once docker-sonic-vs is converted to buster.

**Comment by joyas-joseph** [comment]
retest mellanox please

**Comment by joyas-joseph** [comment]
retest vsimage please

**Comment by lguohan** [comment]
as docker-sonic-vs has been upgrade, can you remove swig?

**Comment by lguohan** [comment]
retest vs please

**Comment by lguohan** [comment]
retest vsimage please

**Comment by joyas-joseph** [comment]
retest vs please

**Comment by joyas-joseph** [comment]
retest mellanox please

**Comment by joyas-joseph** [comment]
retest vs please

**Comment by joyas-joseph** [comment]
retest this please

**Comment by lguohan** [comment]
retest mellanox please

**Comment by pavel-shirshov** [comment]
https://packages.debian.org/buster/swig
buster has required swig. How we get 3.10 version in the logs?
3.10 it is previous version
https://packages.debian.org/stretch/swig
Are we sure we move docker to buster?

**Comment by joyas-joseph** [comment]
> https://packages.debian.org/buster/swig
> buster has required swig version 3.12. How we get 3.10 version in the logs?
> 3.10 it is previous version
> https://packages.debian.org/stretch/swig
> Are we sure we move docker to buster?

libyang was being compiled as a dependency for docker-sonic-vs which was using stretch environment and hence the use of 3.0.10 version of swig.
swig in buster environment is version 3.0.12.

**Comment by qiluo-msft** [comment]
Retest mellanox please

**Comment by joyas-joseph** [comment]
retest mellanox please

**Comment by pavel-shirshov** [comment]
Are we going to use binary which is build under stretch with the binary which is building under buster? Is it safe?

**Comment by joyas-joseph** [comment]
> Are we going to use binary which is build under stretch with the binary which is building under buster? Is it safe?

The way things are setup is that if we have a stretch docker that requires libyang, then it will be built in a stretch environment and that package will be used in the docker. At the same time, if we have a buster docker that requires libyang, then another libyang is built in buster environment and that package would be used.

After the change to use swig from the distribution, we saw a build failure for 'vs' platform only. Build succeeded for other platforms. This was because 'vs' platform had 'docker-sonic-vs' which was a stretch docker and so libyang was being built for stretch. libyang requires at least 3.0.12 version but stretch had 3.0.10 version. Hence the failure.

Now we have docker-sonic-vs converted to buster and so dont see a case of libyang being built for stretch.  Hence the change to add back using swig from the distribution.

**Comment by lguohan** [comment]
retest mellanox please

**Comment by pavel-shirshov** [comment]
Thank you for detailed explanations @joyas-joseph !

**Comment by lguohan** [comment]
it looks like mellanox build keeps failing at the same place.

https://sonic-jenkins.westus2.cloudapp.azure.com/job/mellanox/job/buildimage-mlnx-all-pr/4142/console

**Comment by joyas-joseph** [comment]
> it looks like mellanox build keeps failing at the same place.
> 
> https://sonic-jenkins.westus2.cloudapp.azure.com/job/mellanox/job/buildimage-mlnx-all-pr/4142/console

`make all` for `mellanox` works.

`ENABLE_SYNCD_RPC=y make  target/docker-syncd-mlnx-rpc.gz target/docker-ptf-mlnx.gz` fails with the below message.
```
[ FAIL LOG START ] [ target/debs/stretch/libsaivs-dev_1.0.0_amd64.deb-uninstall ]
target/debs/stretch/libsaivs-dev_1.0.0_amd64.deb does not exist
```

Not sure why `uninstall` is called on something that doesn't exist.

**Comment by joyas-joseph** [comment]
Probably rules for `LIBSAIVS` and `LIBSAIVS_DEV` needs to be moved to `platform/vs/rules.mk`

**Comment by pavel-shirshov** [comment]
Check .dep files.

**Comment by joyas-joseph** [comment]
retest baseimage please

**Comment by lguohan** [comment]
#5039 should solve the mellanox build failure.

**Comment by joyas-joseph** [comment]
retest vsimage please

**Comment by joyas-joseph** [comment]
retest broadcom please

**Comment by lguohan** [comment]
retest broadcom please

**Comment by lguohan** [comment]
retest vsimage please

**Comment by lguohan** [comment]
retest mellanox please
